app/index.js

```diff
@@ -48,115 +48,111 @@ module.exports = yeoman.generators.Base.extend({
             default: "com.foobar.application"
         }];
 
-        this.prompt(prompts, function (props) {
-            props.className = camelcase(props.name);
-            props.slug = decompose(props.name).join("-");
 
-            this.props = props;
-            // To access props later use this.props.someOption;
-
-            done();
-        }.bind(this));
-    },
+        const answers = await this.prompt(prompts)
+        this.props = {};
+        this.props.className = camelcase(answers.name);
+        this.props.slug = decompose(answers.name).join("-");
+        this.props.package = answers.package
+        this.props.name = answers.name
+    }
 
-    writing: {
-        app: function () {
-            // Gradle setup
-            var files = [
-                "build.gradle",
-                "gradle.properties",
-                "gradlew",
-                "gradlew.bat",
-                "settings.gradle",
-                "gradle/idea.gradle",
-                "gradle/repositories.gradle",
-                "gradle/wrapper/gradle-wrapper.properties"
-            ];
-            for (var i = 0; i < files.length; i++) {
-                this.fs.copyTpl(
-                    this.templatePath(files[i]),
-                    this.destinationPath(files[i]),
-                    this.props
-                );
-            }
-            this.fs.copy(
-                this.templatePath("gradle/wrapper/gradle-wrapper.jar"),
-                this.destinationPath("gradle/wrapper/gradle-wrapper.jar")
```

Comment:
We now have to specify all the methods on the base class, NOT as a separate object